sensor/common/deploymentenhancer/component.go

```diff
@@ -76,13 +77,14 @@ func (d *DeploymentEnhancer) Start() error {
 func (d *DeploymentEnhancer) enhanceDeployments(deploymentMsg *central.DeploymentEnhancementRequest) []*storage.Deployment {
 	var ret []*storage.Deployment
 
-	deployments := deploymentMsg.GetMsg().GetDeployments()
-	if deployments == nil {
-		log.Warnf("Received deploymentEnhancement message with no deployments")
+	if deploymentMsg.GetMsg() == nil || deploymentMsg.GetMsg().GetDeployments() == nil {
```

Comment:
In line 64/65 we have `deploymentMsg.GetMsg().GetId()` - if `deploymentMsg.GetMsg()` could theoretically be nil, we are risking a panic there.

Reply:
I don't think so because these are proto message which are `nil` pointer protected:
```go
func (m *DeploymentEnhancementMessage) GetId() string {
	if m != nil {
		return m.Id
	}
	return ""
}

```

Reply:
@fredrb Out of curiosity, I checked in the generated code. Turns out, you both are right 😄 `GetId()` cannot return  `nil`, but `GetMsg()` can definitely.
https://github.com/stackrox/stackrox/blob/mm/21201-sensor-enhance/generated/internalapi/central/deployment_enhancement.pb.go#L140-L145